src/sentry/spans/buffer/redis.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from typing import List
+
+from django.conf import settings
+from sentry_redis_tools.clients import RedisCluster, StrictRedis
+
+from sentry.utils import redis
+
+SEGMENT_TTL = 5 * 60  # 5 min TTL in seconds
+
+
+def get_redis_client() -> RedisCluster | StrictRedis:
+    return redis.redis_clusters.get(settings.SENTRY_SPAN_BUFFER_CLUSTER)
+
+
+class RedisSpansBuffer:
+    def __init__(self):
+        self.client: RedisCluster | StrictRedis = get_redis_client()
+
+    def read_segment(self, segment_id: str) -> List[str | bytes]:
+        segment = self.client.lrange(segment_id, 0, -1)
```

Comment:
You likely want to have a prefix for your key. Like `segments:<segment_id>`. Maybe we should prefix that by the `project_id` as well.